app/models/concerns/translatable.rb

```diff
@@ -47,23 +47,69 @@ def attr_translatable(*attributes)
     end
   end
 
-  def get_translation(attribute)
-    translation = translations.find_by(translatable_attribute: attribute, locale: I18n.locale)
-    translation = get_translations(attribute).first  unless translation
-    return associated_translation(translation) if translation
-    nil
+  def get_translation(attribute, locale: I18n.locale, exact_match: false)
+    # important to filter against the association held in memory instead of performing a db lookup
+    # in order to redisplay potentially transient values
```

Comment:
Sure.  The old implementation was always fetching the data from the db.  But we need now to be able to redisplay the edit step form if there was a validation error and we need to be able to maintain the show the potential edits to the translated values which had not yet been saved to the DB.  I'll expand code comments.